sw/device/tests/sim_dv/flash_rma_unlocked_test.c

```diff
@@ -0,0 +1,298 @@
+};
+const uint32_t kRandomData4[kDataSize] = {
+    0xaf2afb37, 0x646de1b1, 0x092b67d9, 0x1aaaebc3, 0x0dab6bd2, 0x47c83a55,
+    0xbdd1009f, 0xcd9f4c7a, 0xabe6e3a3, 0xb56e9d22, 0xacb955ce, 0x2d05fe1b,
+    0x7e748f0c, 0xc7f4a59d, 0x7cac8713, 0x568ba3c9,
+};
+const uint32_t kRandomData5[kDataSize] = {
+    0x95f9fe5a, 0xb5173233, 0x7055dc6e, 0x9ffdbec7, 0x98d5a883, 0xe81e5751,
+    0x62bbd34e, 0x4bd22ca9, 0x58aac56e, 0xf9d87d26, 0xb2e7eba5, 0x6bcb1c4e,
+    0xec7421c7, 0x8c49708a, 0xbf7932b2, 0x353b6a75,
+};
+const uint32_t kRandomData6[kDataSize] = {
+    0xf8ff57b9, 0x444ff4a9, 0xf3574d3c, 0xf6682a84, 0x67455a38, 0x9c138df2,
+    0x4054e3e5, 0xde4b1a26, 0x047cc121, 0x42cbd0ae, 0x3eec418f, 0x454323c4,
+    0xf21bee28, 0x28dd24b7, 0x29dd06a6, 0xf83e419f,
+};
+const uint32_t kRandomData7[kDataSize] = {
+    0x0eb23677, 0x58c97854, 0x284e1a8f, 0x9b460e99, 0x339b0fe6, 0x80778f39,
+    0x9dbc2981, 0xb4bdc15f, 0x3abbdeb2, 0xab39dd53, 0x96bb2c4a, 0x9b2d1795,
+    0x733bf534, 0xc4914b4b, 0x64487458, 0x9d0fa332,
+};
+
+// RMA unlock token value for LC state transition.
+// TODO : GitHub Issue #11795
+// This is hardcoded to match the token written in the software,
+// when sw_symbol_backdoor_overwrite is fixed to work with ROM
+// targets this will be backdoor written.
+static volatile const uint8_t kLcRmaUnlockToken[LC_TOKEN_SIZE] = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+};
+
+void write_page_scrambled(dif_flash_ctrl_partition_type_t partition_type,
+                          uint32_t page_index, uint32_t region,
+                          const uint32_t *data) {
+  uint32_t address;
+  if (partition_type == kDifFlashCtrlPartitionTypeInfo) {
+    address = flash_ctrl_testutils_info_region_scrambled_setup(
+        &flash_state, page_index, kFlashInfoBank, kPartitionId);
+  } else {
+    address = flash_ctrl_testutils_data_region_scrambled_setup(
+        &flash_state, page_index, region, kRegionSize);
+  }
+
+  CHECK(flash_ctrl_testutils_erase_and_write_page(
+            &flash_state, address, kPartitionId, data, partition_type,
+            kDataSize) == 0);
+}
+
+void read_and_check_page_scrambled(
+    dif_flash_ctrl_partition_type_t partition_type, bool is_equal,
+    uint32_t page_index, uint32_t region, const uint32_t *data) {
+  uint32_t address;
+  uint32_t readback_data[kDataSize];
+  if (partition_type == kDifFlashCtrlPartitionTypeInfo) {
+    address = flash_ctrl_testutils_info_region_scrambled_setup(
+        &flash_state, page_index, kFlashInfoBank, kPartitionId);
+  } else {
+    address = flash_ctrl_testutils_data_region_scrambled_setup(
+        &flash_state, page_index, region, kRegionSize);
+  }
+
+  CHECK(flash_ctrl_testutils_read_page(&flash_state, address, kPartitionId,
+                                       readback_data, partition_type, kDataSize,
+                                       0) == 0);
+
+  for (int i = 0; i < kDataSize; ++i) {
```

Comment:
You can use `CHECK_BUFFER` here. I don't think we have a not-equal version of it but it should be straigtht-forward to write a `CHECK_BUFFER_NOT_EQ` or whatever.